source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContainerView.swift

```diff
@@ -0,0 +1,90 @@
+        backgroundImageViewBottomConstraint.isActive = true
+    }
+    
+    override func addArrangedSubview(_ view: NSView) {
+        super.addArrangedSubview(view)
+        increaseIntrinsicContentSize(view)
+    }
+    
+    override func insertArrangedSubview(_ view: NSView, at insertionIndex: Int) {
+        super.insertArrangedSubview(view, at: insertionIndex)
+        increaseIntrinsicContentSize(view)
+    }
+    
+    override func increaseIntrinsicContentSize(_ view: NSView) {
+        if !view.isHidden {
+            super.increaseIntrinsicContentSize(view)
+            let size = view.intrinsicContentSize
+            if size.width >= 0 && size.height >= 0 {
+                let combinedSize = CGSize(width: max(self.combinedContentSize.width, size.width), height: self.combinedContentSize.height + size.height)
+                self.combinedContentSize = combinedSize
+            }
+        }
+    }
+
+    override func decreaseIntrinsicContentSize(_ view: NSView) {
+        // get max height amongst the subviews that is not the view
+        let maxWidthExcludingTheView = getMaxWidthOfSubviews(afterExcluding: view)
+        let size = getIntrinsicContentSize(inArragedSubviews: view)
+        // there are three possible cases
+        // 1. possibleMaxWidthExcludingTheView is equal to the height of the view
+        // 2. possibleMaxWidthExcludingTheView is bigger than the the height of the view
+        // 3. possibleMaxWidthExcludingTheView is smaller than the the height of the view
+        // only #3 changes the current height, when the view's height is no longer in considreation
+        // for dimension
+        let newWidth = (maxWidthExcludingTheView < size.width) ? maxWidthExcludingTheView : combinedContentSize.width
+        self.combinedContentSize = CGSize(width: newWidth, height: self.combinedContentSize.height - size.height)
+    }
+    
+    override func updateIntrinsicContentSize() {
+        self.combinedContentSize = CGSize.zero
+        super.updateIntrinsicContentSize({ [self] view, id, bool in
+            print(view, id, bool)
+            guard let view = view as? NSView else { return }
+            let size = view.intrinsicContentSize
+            guard !view.isHidden else { return }
+            guard size.width >= 0 && size.height >= 0 else { return }
+            let combinedSize = CGSize(width: max(self.combinedContentSize.width, size.width), height: self.combinedContentSize.height + size.height)
+            combinedContentSize = combinedSize
+        })
+    }
+    
+    func setupBackgroundImageProperties(_ properties: ACSBackgroundImage) {
+        backgroundImageView.fillMode = properties.getFillMode()
+        backgroundImageView.horizontalAlignment = properties.getHorizontalAlignment()
+        backgroundImageView.verticalAlignment = properties.getVerticalAlignment()
+        heightAnchor.constraint(greaterThanOrEqualToConstant: 30).isActive = true
```

Comment:
What is the reason for 30 here?
Is it arbitrary?

Reply:
this method take from ACRColumnView.